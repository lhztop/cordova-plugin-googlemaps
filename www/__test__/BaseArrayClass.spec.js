// jshint esversion: 6
const BaseArrayClass = require('../BaseArrayClass');

describe('[BaseArrayClass]', () => {

  // describe('getLength()', () => {
  //   it('() should return the same size with initial array', () => {
  //     const initArray = [0, 1, 2];
  //     const _ = new BaseArrayClass(initArray);
  //     expect(_.getLength()).toBe(initArray.length);
  //   });
  // });


  describe('mapSeries()', () => {
    it('should error if no callback is given', () => {
      const initArray = ['A', 'B', 'C'];
      const _ = new BaseArrayClass(initArray);
      expect(_.mapSeries(function() {})).rejects.toMatchSnapshot();
    });
    //
    // it('should keep item order', (done) => {
    //   const initArray = ['A', 'B', 'C'];
    //   const _ = new BaseArrayClass(initArray);
    //   let answer = '';
    //   _.mapSeries(function(item, next) {
    //     next(item);
    //   }, function(results) {
    //     console.log(results);
    //     expect(results).toEqual(initArray);
    //     done();
    //   });
    // });
  });

});
