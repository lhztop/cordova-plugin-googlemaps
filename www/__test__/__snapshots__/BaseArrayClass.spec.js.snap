// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[BaseArrayClass] mapSeries() should error if no callback is given 1`] = `[TypeError: Promise.resolve.then is not a function]`;
